webapp/channels/src/utils/post_utils.ts

```diff
@@ -797,6 +797,24 @@ export function getUserOrGroupFromMentionName(
     groupsDisabled?: boolean,
     getMention = getMentionDetails,
 ): [UserProfile?, Group?] {
+    // Handle special case for remote user mentions in the format username:servername
+    if (mentionName.includes(':')) {
+        const parts = mentionName.split(':');
+        const username = parts[0];
+        const serverName = parts[1];
+        if (username && serverName) {
+            // Search for all users with matching username (could be multiple remote users)
+            const userValues = Object.values(users);
+            for (let i = 0; i < userValues.length; i++) {
+                const user = userValues[i];
+                if (user.username === username && user.remote_id === serverName) {
```

Comment:
This is not going to work, as we're comparing the server name to the user's remote ID